codebuild/bin/s2n_open_fds_test.py

```diff
@@ -0,0 +1,66 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script parses the LastDynamicAnalysis file generated by CTest Memcheck.
+# It identifies any leaking file descriptors and triggers an error when detected.
+# This enhances the capabilities of existing Valgrind checks.
+import os
+import re
+import sys
+
+# Exit code matches the valgrind memory loss error code
+ERROR_EXIT_CODE = 123
+
+analysis_file_pattern = re.compile(r"LastDynamicAnalysis.*")
+
+
+def open_analysis_file(path):
+    for fl in os.listdir(path):
+        if analysis_file_pattern.match(fl):
+            file = open(os.path.join(path, fl), 'r')
+    return file
+
+
+def read_analysis_file(file):
+    exit_code = 0
+    to_print = False
+    lines = file.readlines()
+    for i in range(len(lines)):
+        if "<end of output>" in lines[i]:
+            to_print = False
+            continue
+        # Check if the line contains FILE DESCRITOPRS:
+        parts = lines[i].split("FILE DESCRIPTORS: ")
+        if len(parts) > 1:
+            # If a process doesn't leak file descriptors
```

Comment:
I'm a little confused by this logic. Wouldn't it be simpler to just directly assert on the numbers of leaked file descriptors?
```python3
for i, line in enumerate(file.readllines()):
    if not "FILE DESCRIPTORS: " in line:
        continue
    # 1. get the number of file descriptors
    # 2. if the number of file descriptors is > 4 (or whatever the good number is), then 
        # print out the surrounding lines
```

It looks like the current logic will print everything between `FILE DESCRIPTORS` and `<end of output>`. Does that include the file name?

Reply:
> Wouldn't it be simpler to just directly assert on the numbers of leaked file descriptors?

We can't do that. I have stated the reason in the PR description.

> It looks like the current logic will print everything between FILE DESCRIPTORS and <end of output>. Does that include the file name?

It will only print the portion where there are leaking fds. It won't print everything for multi-processes tests.

> if not "FILE DESCRIPTORS: " in line:

This is a simpler and acceptable way. I will use this check to get rid of the last regex that I use.